nifi-nar-bundles/nifi-azure-bundle/nifi-azure-record-sink/src/main/java/org/apache/nifi/services/azure/eventhub/AzureEventHubRecordSink.java

```diff
@@ -0,0 +1,229 @@
+    );
+
+    private volatile ConfigurationContext context;
+    private RecordSetWriterFactory writerFactory;
+    private EventHubProducerClient client;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    protected EventHubProducerClient createEventHubClient(final String namespace, final String serviceBusEndpoint, final String eventHubName,
+                                                          final String policyName, final String policyKey, final AzureAuthenticationStrategy authenticationStrategy) throws ProcessException {
+        final AzureNamedKeyCredential azureNamedKeyCredential = new AzureNamedKeyCredential(policyName, policyKey);
+        final String fullyQualifiedNamespace = String.format("%s%s", namespace, serviceBusEndpoint);
+        final DefaultAzureCredentialBuilder defaultAzureCredentialBuilder = new DefaultAzureCredentialBuilder();
+        final DefaultAzureCredential defaultAzureCredential = defaultAzureCredentialBuilder.build();
+        final EventHubClientBuilder eventHubClientBuilder = new EventHubClientBuilder();
+        if (AzureAuthenticationStrategy.SHARED_ACCESS_KEY == authenticationStrategy) {
+            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, azureNamedKeyCredential);
+        } else {
+            eventHubClientBuilder.credential(fullyQualifiedNamespace, eventHubName, defaultAzureCredential);
+        }
+        return eventHubClientBuilder.buildProducerClient();
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.context = context;
+        writerFactory = context.getProperty(RECORD_WRITER_FACTORY).asControllerService(RecordSetWriterFactory.class);
+        final String namespace = context.getProperty(NAMESPACE).evaluateAttributeExpressions().getValue();
+        final String serviceBusEndpoint = context.getProperty(SERVICE_BUS_ENDPOINT).evaluateAttributeExpressions().getValue();
+        final String eventHubName = context.getProperty(EVENT_HUB_NAME).evaluateAttributeExpressions().getValue();
+        final String policyName = context.getProperty(SHARED_ACCESS_POLICY).getValue();
+        final String policyKey = context.getProperty(SHARED_ACCESS_POLICY_KEY).getValue();
+        final String authenticationStrategy = context.getProperty(AUTHENTICATION_STRATEGY).getValue();
+        final AzureAuthenticationStrategy azureAuthenticationStrategy = AzureAuthenticationStrategy.valueOf(authenticationStrategy);
+        client = createEventHubClient(namespace, serviceBusEndpoint, eventHubName, policyName, policyKey, azureAuthenticationStrategy);
+    }
+
+    @OnDisabled
+    public void onDisabled() {
+        if (client == null) {
+            getLogger().debug("Event Sender not configured");
+        } else {
+            client.close();
+        }
+    }
+
+    // put all records in one event, then see capacity of event
```

Comment:
This should be changed to a JavaDoc comment, or removed.
```suggestion
    /**
     * Send each Record as an Event Data object
     */
```